core/templates/packages.el.template

```diff
@@ -1,31 +1,63 @@
-;;; packages.el --- %LAYERNAME% Layer packages File for Spacemacs
+;;; packages.el --- %LAYER_NAME% layer packages file for Spacemacs.  -*- lexical-binding: t; -*-
```

Comment:
I'm happy to relent on this one if it's a blocker on the PR!

For your interest, as I understand it, lexical binding in elisp changes two things:
1. changes `let` to to bind variables with lexical scope
2. changes `lambda` to capture variables in the enclosing scope.

This is what every modern language with closures does, so I think it's a huge improvement. But I can imagine situations where contributors copy-pasta stuff off emacswiki that depends on dynamic scope and breaks. So I'm happy either way. :)